test/test_ufbrw.F90

```diff
@@ -0,0 +1,180 @@
+
+      print *, 'Testing misc subroutines.'
+
+#ifdef KIND_8
+      call setim8b(.true.)
+#endif
+
+! set the name of the bufr testfile to process
+
+      file="testfiles/data/prepbufr2.ref"
+
+! open the print file to record ufbrep test results
+
+      open(55,file='ufbrw_prnt_out')
+
+! test various reading options applying user specified filtering, testing conwin
+
+      do i=1,5
+
+      if(i==1) cond='POB<1000         '
+      if(i==2) cond='POB<1000 UOB<1000'
+      if(i==3) cond='POB<800  TOB<1000'
+      if(i==4) cond='POB<851  POB>699 '
+      if(i==5) cond='POB=850          '
+      write(55,*) cond
+
+! set the filename and open and read the testfile with various conditions specified
+! print the results in a text file for verification
+
+      open(20,file=file,form='unformatted')
+      call openbf(20,'IN',20)
+
+      do while(ireadmg(20,subset,idate)==0)
+      do while(ireadsb(20)==0)
+      call ufbint(20,arr,10,255,irt,cond//' POB QOB TOB UOB VOB')
+      if(irt>0) write(55,'(5(1x,f8.2))')arr(1:5,1:irt)
+      enddo
+      enddo
+      call closbf(20)
+      write(55,*)
+      enddo
+
+! copy the testfile to test helpers getwin, invwin, newwin, nxtwin in output more
+! print the results in a text file for verification
+
+      open(20,file=file,form='unformatted')
+      open(50,file='ufbrw_bufr_out',form='unformatted')
+
+      call openbf(20,'IN ',20)
+      call openbf(50,'OUT',20)
+      
+      write(55,*);write(55,*)'read/write from unit 20'
+
+      do while(ireadmg(20,subset,idate)==0)
+      do while(ireadsb(20)==0)
+      call ufbint(20,arr,10,255,irt,'POB QOB TOB UOB VOB')
+      write(55,'(5(1x,f8.2))')arr(1:5,1:irt)
+
+      call openmb(50,subset,idate)
+      call ufbint(50,arr,10,irt,jrt,'POB QOB TOB UOB VOB')
+      call writsb(50)
+      enddo
+      enddo
+
+! read the testfile copy and print the results in a text file for verification
+
+      call closbf(50)
+      open(50,file='ufbrw_bufr_out',form='unformatted')
+      call openbf(50,'IN',50) 
+
+      write(55,*);write(55,*)'read/write from unit 50'
+
+      do while(ireadmg(50,subset,idate)==0)
+      do while(ireadsb(50)==0)
+      call ufbint(50,arr,10,255,irt,'POB QOB TOB UOB VOB')
+      write(55,'(5(1x,f8.2))')arr(1:5,1:irt)
+      enddo
+      enddo
+
+! close the testfile for writing and open it for reading
+
+      close(55)
+      open(55,file='ufbrw_prnt_out')
+
+! verify the testfile contents against output stored in brr array strings
+
+      do n=1,55  
+      read(55,'(a55)',iostat=iret) line
+      call strsuc(line  ,str1,len1)
+      call strsuc(brr(n),str2,len2)
+      if(n<=55.and.iret==0.and.str1/=str2) then
+         print*,n,len1,len2
+         print*,"str1:",str1  
+         print*,"str2:",str2 
+         call bort('error exit from ufbrw_test')
+      elseif(n>55.and.iret==0) then
+         print*,n,iret
+         call bort('error exit from ufbrw_test')
+      elseif(n==56) then
```

Comment:
What's this elsif for? Won't the do loop already end without it when n is 56?

Reply:
checking to see if the copy made to many lines.